CMSgov/design-system
# [WNMGDS-971] Convert Drawer and HelpDrawer to Typescript

## Summary
[Jira ticket](https://jira.cms.gov/browse/WNMGDS-971)

Converted Drawer, DrawerToggle, HelpDrawer, and HelpDrawer toggle to Typescript function components. 

## How to test
- Check if it builds: `yarn build`
- Check if it starts: `yarn start`
- Make sure Storybook starts: `yarn storybook`
- Mark sure TS checks pass: `yarn type-check`

On all site builds, navigate to the Drawer component and check console for errors.
